src/ucai/core/databricks.py

```diff
@@ -84,6 +89,15 @@ def _try_get_spark_session_in_dbr() -> Any:
         return
 
 
+def _is_in_databricks_notebook_environment() -> bool:
+    try:
+        from dbruntime.databricks_repl_context import get_context
+
+        return get_context().isInNotebook
+    except (ImportError, AttributeError):
```

Comment:
What about other exceptions 🤔 This is internal function so if there're other exceptions there's nothing users can do. Maybe we just always return False when there's exception?

Reply:
Excellent point! broadened the scope